src/account.c

```diff
@@ -1420,6 +1421,91 @@ const char *account_extra(const struct account *acc)
 }
 
 
+/**
+ * Auto complete a SIP uri, add scheme and domain if missing
+ *
+ * @param acc User-Agent account
+ * @param buf Target buffer to print SIP uri
+ * @param uri Input SIP uri
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int account_uri_complete(const struct account *acc, struct mbuf *buf,
+			 const char *uri)
+{
+	struct sa sa_addr;
+	size_t len;
+	bool uri_is_ip;
+	char *uridup;
+	char *host;
+	char *p;
+	int err = 0;
+
+	if (!buf || !uri)
+		return EINVAL;
```

Comment:
also check `acc` here

Reply:
`acc` is checked later, because it can be `NULL`.  see usage in `uag_find_requri()`, where `acc` is not known. if `acc` is `NULL`. missing "sip:" is appended anyway.